app/questionnaire/routing/when_rule_evaluator.py

```diff
@@ -19,19 +19,20 @@ class WhenRuleEvaluator:
     list_store: ListStore
     metadata: Mapping
     response_metadata: Mapping
-    location: Union[Location, RelationshipLocation]
+    location: Optional[Union[Location, RelationshipLocation]]
```

Comment:
As per the style guide, optional when we have one type. 
```suggestion
    location: Union[None, Location, RelationshipLocation]
```